src/main/metadata.ts

```diff
@@ -0,0 +1,54 @@
+const ROW_ID_KEY: string = "rowId";
+
+/**
+ * Retrieves the metadata object for a given range. If the metadata object does not exist,
+ * this function will set the row id key so the metadata object is created and a reference
+ * can be held
+ * 
+ * @param range Range representing an event row
+ * @returns reference to the Metadata object for this row
+ */
+export function getMetadata(range: Range): Metadata {
+    let metadataList: Metadata[] = range.getDeveloperMetadata();
+
+    if (metadataList.length === 0) {
+        // Add the row id key so the metadata object is created
+        range.addDeveloperMetadata(ROW_ID_KEY);
+        metadataList = range.getDeveloperMetadata();
+    }
+
+    return metadataList[0];
+}
+
+/**
+ * Retrieves the unique id for a Row. The unique id is stored in the value of the metadata.
+ * If the unique id has not been set, null will be returned
+ * 
+ * @param row the Row to retrieve the unique id for
+ * @returns unique id for the row; null if one has not been set
+ */
+export function getRowUniqueId(row: Row): string | null {
```

Comment:
non-blocking comment: in my mind "id" is usually unique, so i'm wondering if dropping "unique" from these function names would help with conciseness. not 100% sure of this suggestion but if we were to also infer we're working with rows b/c of the input argument, we could even call them "getId", "generateId", "hasId" etc.

Reply:
Will drop Row and Unique from the function names